### PR TITLE
caf: events: Add documentation for power events

### PR DESCRIPTION
**Pull request by MarekPieta**
Change adds doxygen documentation for power events.

Jira: NCSDK-9551

### CODE REVIEW

**Review by pdunaj — APPROVED** [review]

**Comment by MarekPieta** [comment]
Retriggered CI

File: include/caf/events/power_event.h [hunk 1]
```diff
@@ -19,13 +20,59 @@
...
+ * to be active after the fatal error. For example, module that controls hardware LEDs may be used
+ * to display LED effect related to the fatal error.
+ *
+ * The module that controls power management is the final subscriber for the power down event.
+ * When this module receives the power down event, it's ensured that all other application modules
+ * that handle power down event are already suspended. Then power down procedure can be continued.
+ *
+ * @warning An application module that handles power down event must also handle @ref wake_up_event.
+ *          Otherwise the module will never be woken up after suspending.
+ */
```
Comment (emob-nordic) [inline include/caf/events/power_event.h:50]: Discussed offline:
I think we should not describe in detail what power management module should look like and how modules should submit/react to `wake_up/power_down _event`. Here we should only mention what the events are. All additional information would be put to other place.

This part of documentation should be shortened to something like:
The power down event is used to inform application modules that each of them should enter suspended mode, where power consumption is reduced and for modules to signalize fatal error.

File: include/caf/events/power_event.h [hunk 2]
```diff
@@ -19,13 +20,59 @@
...
+ * sensor trigger.
+ *
+ * Every application module that handles wake up event should exit suspended state and turn on
+ * functionalities disabled on @ref power_down_event. The module must not consume the wake up event
+ * to ensure that other listeners will be informed too.
+ *
+ * The module that controls power management of the application should be an only early subscriber
+ * for the wake up event. This module can consume the wake up events to prevent waking up the
+ * application modules after fatal error.
+ */
```
Comment (emob-nordic) [inline include/caf/events/power_event.h:73]: More or less the same here. Right now the documentation describes an action at first place, not what the event is. The exact explanation should not be a part of this event documentation.

**Review by emob-nordic — CHANGES_REQUESTED** [review]

File: include/caf/events/power_event.h (hunk 1 above)
Comment (MarekPieta) [reply include/caf/events/power_event.h:50]: Added this at the beginning of the event's documentation

File: include/caf/events/power_event.h (hunk 2 above)
Comment (MarekPieta) [reply include/caf/events/power_event.h:73]: Changed

**Review by emob-nordic — APPROVED** [review]

**Pull request merged**